app/api/user_endpoint.py

```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
     """
     Sets user preference.Uses key to set the preference
     """
-    pass
+    user_preference = storage.driver.user.preference[key]
```

Comment:
storage.driver.user does not seem to have preference attribute with __getitem__ dunder, you get user from args